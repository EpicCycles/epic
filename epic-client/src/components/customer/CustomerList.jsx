import React from 'react'
import {Dimmer, Loader} from 'semantic-ui-react'
import {Redirect} from "react-router-dom";
import CustomerListAndSearch from "./CustomerListAndSearch";

class CustomerList extends React.Component {
    state = {};

    goToAddCustomer = () => {
        this.props.clearCustomerState();
        this.setState({ redirect: '/customer' });
    };

    goToEditCustomer = (customerId) => {
        this.props.getCustomer(customerId);
        this.setState({ redirect: '/customer' });
    };

    render() {
        const { redirect } = this.state;
        const { getCustomerList, getCustomerListPage, getCustomer, isLoading, customers, count, next, previous, searchParams } = this.props;
        if (redirect) return <Redirect to={redirect}/>;

        return (
            <div id="customer-list" className="grid-container">
                <CustomerListAndSearch
                    addNewCustomer={this.goToAddCustomer}
                    getCustomerList={getCustomerList}
                    getCustomerListPage={getCustomerListPage}
                    getCustomer={this.goToEditCustomer}
                    searchParams={searchParams}
                    isLoading={isLoading}
                    customers={customers}
                    count={count}
                    next={next}
                    previous={previous}
                />
                {isLoading &&
                <Dimmer active inverted>
                    <Loader content='Loading'/>
                </Dimmer>
                }
            </div>
        )
    }
}

export default CustomerList;