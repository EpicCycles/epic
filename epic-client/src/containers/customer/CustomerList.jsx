import React from 'react'
import {Button, Dimmer, Loader} from 'semantic-ui-react'
import Pagination from "../../common/pagination";
import FormTextInput from "../../common/FormTextInput";
import CustomerRow from "./CustomerRow";
import ErrorDismissibleBlock from "../../common/ErrorDismissibleBlock";

class CustomerList extends React.Component {
    state = {
        firstName: '',
        lastName: '',
    };

    componentWillMount() {
        if (this.props.searchLastName || this.props.searchFirstName) {
            this.setState({
                lastName: this.props.searchFirstName,
                firstName: this.props.searchLastName
            });
        }
    }

    onFirstNameChanged = input => {
        this.setState({ firstName: input });
    };

    onLastNameChanged = input => {
        this.setState({ lastName: input });
    };

    onClearFirstName = () => {
        this.setState({ firstName: '' });
    };

    onClearLastName = () => {
        this.setState({ lastName: '' });
    };

    onSubmit= (event) => {
        event.preventDefault();
        event.stopPropagation();
        const {firstName, lastName} = this.state;
        this.props.getCustomerList(firstName, lastName);
    };

    render() {
        const { firstName, lastName } = this.state;
        const {getCustomerListPage, getCustomer,removeCustomerError, isLoading, customers, count, page, totalPages, error} = this.props;

        return (
            <div id="customer-list">
                <form onSubmit={this.onSubmit}>
                <h1>Find Customer</h1>
                <label htmlFor="search-input" className="search-heading" id="search-input-label">
                    Please enter your search criteria.
                </label>
                <div className="row">
                    <FormTextInput
                        placeholder="First Name"
                        id="first-name-input"
                        className="column full"
                        onChange={this.onFirstNameChanged}
                        value={firstName}
                        onClick={this.onClearFirstName}
                    />
                </div>
                <div className="row">
                    <FormTextInput
                        placeholder="Last Name"
                        id="last-name-input"
                        className="column full"
                        onChange={this.onLastNameChanged}
                        value={lastName}
                        onClick={this.onClearLastName}
                    />
                </div>
                <p>
                    <Button type="submit" disabled={isLoading}>Get Customers</Button>
                </p>
                </form>
                {count > 0 ?
                    <div style={{width: '100%', height: '400px'}}>
                        <table className="fixed_headers">
                            <thead>
                            <tr>
                                <th className="listHead">First Name</th>
                                <th className="listHead">Last Name</th>
                                <th className="listHead">email</th>
                                <th className="listHead">Date Added</th>
                                <th className="listHead">Date Updated</th>
                                <th className="listHead">Actions</th>
                            </tr>
                            </thead>
                            <tbody>
                            {customers.map(customer =>
                                <CustomerRow key={`cust-${customer.id}`} customer={customer} getCustomer={getCustomer}/>
                            )}
                            </tbody>
                        </table>
                        <Pagination
                            id="customer-pagination"
                            page={page}
                            totalPages={totalPages}
                            getPage={getCustomerListPage}/>
                    </div>
                    :
                    <p>
                        Set criteria and search.
                    </p>
                }
                {error &&
                <ErrorDismissibleBlock error={error} removeError={removeCustomerError} />
                }

                {isLoading &&
                <Dimmer active inverted>
                    <Loader content='Loading'/>
                </Dimmer>
                }
            </div>
        )
    }
}

export default CustomerList;