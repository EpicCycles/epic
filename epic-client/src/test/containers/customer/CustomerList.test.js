import React from 'react';
import configureStore from 'redux-mock-store'
import CustomerList from "../../../containers/customer/CustomerList";


describe("CustomerList.index tests", () => {
    // our mock login function to replace the one provided by mapDispatchToProps
    const mockGetCustomerListAsyncfn = jest.fn();
    const mockGetCustomerListPagefn = jest.fn();
    const initialState = {customer:{count: 0,
        isLoading: false,
        customers: [],
        page: 1,
        perPage: 5}};


    it('renders the CustomerList correctly', () => {
        let initialList = shallow(<CustomerList
            getCustomerListAsync={mockGetCustomerListAsyncfn}
            getCustomerListPage={mockGetCustomerListPagefn}
            count={0}
            customers={[]}
            isLoading={false}
            page={0}
            totalPages={0}
        />);
        expect(initialList).toMatchSnapshot();
    });
});
