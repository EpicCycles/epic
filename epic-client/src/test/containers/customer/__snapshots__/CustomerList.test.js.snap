// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerList.index tests renders the CustomerList correctly 1`] = `
<div
  id="customer-list"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Find Customer
    </h1>
    <label
      className="search-heading"
      htmlFor="search-input"
      id="search-input-label"
    >
      Please enter your search criteria.
    </label>
    <div
      className="row"
    >
      <FormTextInput
        className="column full"
        id="first-name-input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="First Name"
        value=""
      />
    </div>
    <div
      className="row"
    >
      <FormTextInput
        className="column full"
        id="last-name-input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Last Name"
        value=""
      />
    </div>
    <p>
      <Button
        as="button"
        disabled={false}
        role="button"
        type="submit"
      >
        Get Customers
      </Button>
    </p>
  </form>
  <p>
    Set criteria and search.
  </p>
</div>
`;
